packages/scoutgame/src/builderNfts/refreshEstimatedPayouts.ts

```diff
@@ -0,0 +1,93 @@
+import { prisma } from '@charmverse/core/prisma-client';
+import type { ISOWeek } from '@packages/dates/config';
+import { getCurrentSeasonStart } from '@packages/dates/utils';
+
+import { nftTypeMultipliers } from '../points/dividePointsBetweenBuilderAndScouts';
+import { getPointsCountForWeekWithNormalisation } from '../points/getPointsCountForWeekWithNormalisation';
+
+import type { BuilderNftWithOwners, SeasonNfts } from './getAllSeasonNftsWithOwners';
+import { getAllSeasonNftsWithOwners } from './getAllSeasonNftsWithOwners';
+
+/**
+ * Refresh estimated payouts for a given week
+ *
+ * @param builderToRefresh - Builder ID to refresh. If not provided, all builders will be refreshed
+ */
+export async function refreshEstimatedPayouts({
+  week,
+  builderIdToRefresh
+}: {
+  week: ISOWeek;
+  builderIdToRefresh?: string;
+}): Promise<SeasonNfts> {
+  const season = getCurrentSeasonStart(week);
+
+  const { normalisedBuilders } = await getPointsCountForWeekWithNormalisation({
+    week
+  });
+
+  const data = await getAllSeasonNftsWithOwners({ season });
+
+  const seasonBuilderNfts = data.default.reduce(
+    (acc, nft) => ({
+      ...acc,
+      [nft.builderId]: nft
+    }),
+    {} as Record<string, BuilderNftWithOwners>
+  );
+
+  const seasonStarterPackNfts = data.starter_pack.reduce(
+    (acc, nft) => ({
+      ...acc,
+      [nft.builderId]: nft
+    }),
+    {} as Record<string, BuilderNftWithOwners>
+  );
+
+  for (const { builder, normalisedPoints } of normalisedBuilders) {
+    if (!builderIdToRefresh || builderIdToRefresh === builder.builder.id) {
+      const defaultNft = seasonBuilderNfts[builder.builder.id];
+      const starterPackNft = seasonStarterPackNfts[builder.builder.id];
+
+      const defaultNftBalance = defaultNft.nftOwners.reduce((acc, nft) => acc + nft.balance, 0);
+      const starterPackNftBalance = starterPackNft?.nftOwners.reduce((acc, nft) => acc + nft.balance, 0) ?? 0;
+
+      const builderNftBalanceWeighted = defaultNftBalance * nftTypeMultipliers.default;
+      const starterPackNftBalanceWeighted = starterPackNftBalance * nftTypeMultipliers.starter_pack;
+
+      const totalNftBalance = builderNftBalanceWeighted + starterPackNftBalanceWeighted;
+
+      const expectedPayoutForNextNftPurchase = Math.floor(
+        normalisedPoints / (totalNftBalance + nftTypeMultipliers.default)
```

Comment:
maybe just a comment here to remind the reader why we have to look up nftTypeMultipliers, like "Use value from nftTypeMultipliers.default to increase the nft balance by one"?

The variable name multiplier doesnt mean anything to me and it's weird we are adding it